sys/trickle/trickle.c

```diff
@@ -30,63 +27,61 @@ void trickle_callback(trickle_t *trickle)
     if ((trickle->c < trickle->k) || (trickle->k == 0)) {
         (*trickle->callback.func)(trickle->callback.args);
     }
+
+    trickle_interval(trickle);
 }
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t max_interval;
+    uint32_t old_interval = trickle->I;
```

Comment:
in our current trickle implementation we calculate the `future I`, i.e. the `I` of the next iteration isntead of the current iteration. This way, and by calculating the diff of `t` and the next `I`, we can spare one timer/msg. But for this, we need to save the current `I`, before it's capped by `Imax`.